lib/sort-order-selector/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { toggleSortOrder, setSortType } from '../state/settings/actions';
+import classNames from 'classnames';
+import IconButton from '../icon-button';
+import SortOrderIcon from '../icons/sort-order';
+
+import * as S from '../state';
+import * as T from '../types';
+
+type StateProps = {
+  shouldDisplay: boolean;
+  sortReversed: boolean;
+  sortType: T.SortType;
+};
+
+type DispatchProps = {
+  toggleSortOrder: () => any;
+  setSortType: (sortType: T.SortType) => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export class SortOrderSelector extends Component<Props> {
+  static displayName = 'SortOrderSelector';
+
+  changeSort = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.props.setSortType(event.currentTarget?.value);
+  };
+
+  render() {
+    const { shouldDisplay, sortReversed, sortType } = this.props;
+
+    let sortLabel = sortReversed ? 'Oldest' : 'Newest';
+    if (sortType === 'alphabetical') {
+      sortLabel = sortReversed ? 'Z-A' : 'A-Z';
+    }
+
+    const sortTypes = [
+      {
+        label: 'Date modified',
+        id: 'modificationDate',
+      },
+      {
+        label: 'Date created',
+        id: 'creationDate',
+      },
+      {
+        label: 'Alphabetical',
+        id: 'alphabetical',
+      },
+    ];
+
+    return (
+      <Fragment>
+        {shouldDisplay && (
+          <div className="sort-order-selector">
+            Sort:
```

Comment:
not that we're generally that accessible, but would it be worth surrounding this in a `<label for="sort-selection">` tag so that there's a link between it and the dropdown?